### PR TITLE
[iOS] Implement burning AI chats for single tab

### PR DESCRIPTION
**Pull request by hassaanelgarem**
Task/Issue URL: https://app.asana.com/1/137249556945/project/1211767540372501/task/1212775673257400?focus=true
CC: @dus7 @aataraxiaa @amddg44  

### Description
This PR implements chat deletion when burning a single tab. This includes:
- Deleting the current chat when burning a single tab that contains an AI Chat
- Deleting contextual AI chats when burning data for tabs that have an active contextual chat session
- Implement chat deletion syncing via `AIChatSyncCleaner` to notify sync servers of deleted chats

### Testing Steps

0. Enable `aiChatSync`, `burnSIngleTab`, `contextualDuckAIMode` 
1. Open an AI Chat tab and start a conversation
2. Use the fire button and burn the tab
3. Open another AI Chat tab and check the chats list
4. Verify the specific chat is deleted
5. Open a regular tab and start a contextual chat by tapping the duck ai button
6. Burn the tab's data and verify the contextual chat is also deleted from the chats list
7. With sync enabled, verify deleted chats are synced across devices

### Impact and Risks
Medium: Could disrupt specific features or user flows

#### What could go wrong?
- Chat deletion could fail if the headless WebView is already in use
- Sync deletion could fail if the network is unavailable (mitigated by persisting chat IDs to delete and retrying on next sync)

### Quality Considerations
- Added comprehensive unit tests for `FireExecutor` covering single chat deletion, contextual chat deletion, and edge cases
- Added `AIChatSyncCleanerTests` for sync chat deletion functionality
- Added `TabViewModelTests` for chat ID extraction
- Sync deletion uses a pending queue to retry failed deletions

### Notes to Reviewer

- This PR builds on top of #3360 which contains the single-tab data burning foundation
- The `historyCleanerProvider` closure pattern was introduced to allow fresh `HistoryCleaner` instances per operation (avoiding "operation in progress" conflicts)
- Single tab AI chat burning intentionally bypasses the `autoClearAIChatHistory` setting since it's an explicit user action on a specific tab
- Contextual chat burning respects the `autoClearAIChatHistory` setting.
---
###### Internal references:
[Definition of Done](https://app.asana.com/0/1202500774821704/1207634633537039/f) | [Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552) | [Tech Design Template](https://app.asana.com/0/59792373528535/184709971311943)

---

> [!NOTE]
> **Medium Risk**
> Adds new per-chat deletion paths (local JS-based clearing + server-side sync) and changes Fire tab-scoped burn behavior, which could cause missed or unintended deletions if chat IDs are wrong or operations overlap.
> 
> **Overview**
> Enables **single AI chat deletion** when burning a *single tab*: `FireExecutor` now extracts a `chatID` from the tab URL and calls `HistoryCleaner.deleteAIChat`, bypassing the `autoClearAIChatHistory` setting for this explicit tab action.
> 
> Adds **contextual chat cleanup** during tab-scoped *data* burns: if `autoClearAIChatHistory` is enabled and a tab has an active contextual chat ID, the chat is deleted and the contextual sheet state is cleared via `AIChatContextualSheetCoordinator.clearActiveChat`.
> 
> Extends the AI Chat clearing pipeline to support per-chat deletion by passing an optional `chatId` parameter into the JS message (`AIChatDataClearingUserScript`/`HistoryCleaner`), guarding against concurrent operations (`operationInProgress`), and introducing sync-side deletion by chat IDs with persistence/retry (`AIChatSyncCleaner`, new `DDGSyncing.deleteAIChats(chatIds:)` + `AIChatsHandling.delete(chatIds:)`). Tests are expanded across AIChat sync cleaner, Fire tab burn behavior, contextual coordinator reset, and `TabViewModel` chat ID extraction.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 67ae2545f0291e779302ae9a768aade437e5e4e6. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: SharedPackages/AIChat/Sources/AIChat/Shared/HistoryCleaner/HistoryCleaner.swift [hunk 1]
```diff
@@ -45,18 +46,35 @@ public final class HistoryCleaner: HistoryCleaning {
...
+    public func deleteAIChat(chatID: String) async -> Result<Void, Error> {
+        guard webView == nil else {
+            return .failure(HistoryCleanerError.operationInProgress)
+        }
 
         return await withCheckedContinuation { continuation in
             self.continuation = continuation
             Task { @MainActor in
-                await self.processAllDomains()
+                await self.processAllDomains(chatID: chatID)
```
[the remark this answers is not in the document]
Comment (hassaanelgarem) [reply SharedPackages/AIChat/Sources/AIChat/Shared/HistoryCleaner/HistoryCleaner.swift:66]: 4af6cb177b954af1d53799077f68e65376afcdb8

File: SharedPackages/BrowserServicesKit/Sources/DDGSync/internal/AIChats.swift [hunk 2]
```diff
@@ -53,4 +54,31 @@ public final class AIChats: AIChatsHandling {
             throw SyncError.unexpectedStatusCode(statusCode)
         }
     }
+
+    public func delete(chatIds: [String], token: String) async throws {
+        guard !chatIds.isEmpty else { return }
+
+        let updates: [[String: String]] = chatIds.map { chatId in
+            ["id": chatId, "deleted": "true"]
+        }
```
[the remark this answers is not in the document]
Comment (hassaanelgarem) [reply SharedPackages/BrowserServicesKit/Sources/DDGSync/internal/AIChats.swift:63]: It is working fine now, and passing an empty string actually doesn't work

File: SharedPackages/AIChat/Sources/AIChat/Shared/Sync/AIChatSyncCleaner.swift [hunk 3]
```diff
@@ -126,6 +141,21 @@ public final class AIChatSyncCleaner: AIChatSyncCleaning {
...
+            return
+        }
+
+        do {
+            try await sync.deleteAIChats(chatIds: pendingChats)
+            await state.clearChatsToBeDeleted()
+        } catch {
+            Logger.aiChat.debug("Failed to delete pending ai chats: \(error.localizedDescription)")
+        }
+    }
```
[the remark this answers is not in the document]
Comment (hassaanelgarem) [reply SharedPackages/AIChat/Sources/AIChat/Shared/Sync/AIChatSyncCleaner.swift:160]: 801b78e04901413aca97f2e8bfc6c3e03409aaff 👍
